### PR TITLE
simplified import logic

### PR DESCRIPTION
**Pull request by SustainedCruelty**
I simplified some of the code for importing json and yaml files.
Instead of saving the flags before unmarshalling, we can simply parse the flags again after we've read in the file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -104,12 +104,28 @@ var generateCmd = &cobra.Command{
...
 			if err != nil {
 				return err
 			}
+			defer f.Close()
+
+			if strings.HasSuffix(importPath, ".json") {
+				if err = json.NewDecoder(f).Decode(&file); err != nil {
+					return err
+				}
+			} else if strings.HasSuffix(importPath, ".yaml") && strings.HasSuffix(importPath, ".yml") {
```
Comment (hopefulTex) [inline main.go:117]: This line requires both the `.yml` tag and the `.yaml` tag.
It should be `or`

Comment (maaslalani) [reply main.go:117]: Good catch @hopefulTex!

Comment (maaslalani) [inline main.go:117]:
```suggestion
			} else if strings.HasSuffix(importPath, ".yaml") || strings.HasSuffix(importPath, ".yml") {
```

**Comment by hopefulTex** [comment]
How does this work?  If the flags aren't parsed until after the import file is read then how does the program know what the import path flag is set to?
Does it parse internally and the `.Parse()` call just does so again?

**Pull request closed without merging**